RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripRawToClusterAlgo.dev.cc

```diff
@@ -0,0 +1,1096 @@
+      throw cms::Exception("SiStripRawToClstAlg", "MaxClusterSize must be <= 768");
+    }
+  }
+
+  void SiStripRawToClusterAlgo::prepareUnpackCluster(Queue& queue,
+                                                     const DetToFeds* conditions_DetToFeds,
+                                                     std::unique_ptr<PortableFEDMover> rFEDChMover) {
+    // Move ownership of the host-data container this class
+    fedChMover_ = std::move(rFEDChMover);
+
+    // Move the data to the device
+    fedBuffer_d_.emplace(cms::alpakatools::make_device_buffer<uint8_t[]>(queue, fedChMover_->bufferSize()));
+    alpaka::memcpy(queue, *fedBuffer_d_, fedChMover_->buffer(), fedChMover_->bufferSize());
+
+    // Move fedchannels to the device
+    stripMapping_d_ = cms::alpakatools::moveToDeviceAsync(queue, fedChMover_->mapping());
+
+    // Apply quality conditions to mapping and calculate the number of strips to unpack
+    uint32_t divider = 32u;
+    uint32_t groups = divide_up_by(fedChMover_->channelNb(), divider);
+    auto workDiv = make_workdiv<Acc1D>(groups, divider);
+
+    auto nstrips_d = make_device_buffer<uint32_t>(queue);
+    alpaka::memset(queue, nstrips_d, 0);
+
+    auto invalidFedChN_d = make_device_buffer<uint32_t>(queue);
+    alpaka::memset(queue, invalidFedChN_d, 0);
+
+    alpaka::exec<Acc1D>(queue,
+                        workDiv,
+                        SiStripKer_applyQualConds{},
+                        nstrips_d.data(),
+                        invalidFedChN_d.data(),
+                        fedBuffer_d_->data(),
+                        stripMapping_d_->view(),
+                        conditions_DetToFeds);
+
+    const uint32_t threads = 1024u;
+    const uint32_t nBlocks = divide_up_by(stripMapping_d_->view().metadata().size(), threads);
+    const auto workDivMultiBlock = make_workdiv<Acc1D>(nBlocks, threads);
+    auto blockCounter_d = make_device_buffer<int32_t>(queue);
+    alpaka::memset(queue, blockCounter_d, 0);
+    alpaka::exec<Acc1D>(queue,
+                        workDivMultiBlock,
+                        multiBlockPrefixScan<uint32_t>(),
+                        stripMapping_d_->const_view().fedChStripsN().data(),
+                        stripMapping_d_->view().fedChStripsN().data(),
+                        (uint32_t)stripMapping_d_->view().metadata().size(),
+                        (int32_t)nBlocks,
+                        blockCounter_d.data(),
+                        alpaka::getPreferredWarpSize(alpaka::getDev(queue)));
+
+    // To do: merge the two kernels for applyQualConds and sum into a single one.
+
+    // Get the number of strips to unpack
+    nStrips_h_ = cms::alpakatools::make_host_buffer<uint32_t>(queue);
+    const uint32_t size = stripMapping_d_->view().metadata().size();
+    auto viewSrc = make_device_view(queue, stripMapping_d_->view().fedChStripsN(size - 1));
+    auto viewDst = make_host_view(*nStrips_h_->data());
+    alpaka::memcpy(queue, viewDst, viewSrc);
```

Comment:
```suggestion
    alpaka::memcpy(queue, *nStrips_h_, viewSrc);
```
`*nStrips_h_` is an alpaka buffer, and can be used directly in a `memcpy`.